### PR TITLE
TEST/APPS: Improve IO demo

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Improve IO demo test application

## Why ?

Make it more configurable/robust

## How ?

Fixes, cleanups and new parameters

### CODE REVIEW

**Comment by dmitrygx** [comment]
bot:pipe:retest

**Comment by dmitrygx** [comment]
@brminich @hoopoepg @evgeny-leksikov @alinask could you review pls?

File: contrib/test_jenkins.sh [hunk 1]
```diff
@@ -902,7 +902,7 @@ run_io_demo() {
 
 	echo "==== Running UCP IO demo  ===="
 
-	test_args="$@"
+	test_args="$@ -o write,read -d 128:4194304 -i 10000 -w 10"
```
Comment (yosefe) [inline contrib/test_jenkins.sh:905]: 👍 good addition

File: test/apps/iodemo/ucx_wrapper.cc [hunk 2]
```diff
@@ -99,494 +102,613 @@ UcxContext::UcxContext(size_t iomsg_size, bool verbose) :
...
-                 << std::endl;
-    remove_connection(conn);
-    conn->disconnect();
-    if (!conn->is_disconnected()) {
-        _closing_conns.push_back(conn);
-    }
+    ucx_request *r = reinterpret_cast<ucx_request*>(request);
+    r->completed = true;
+    r->conn_id   = (info->sender_tag & ~IOMSG_TAG) >> 32;
+    r->recv_length    = info->length;
```
Comment (yosefe) [inline test/apps/iodemo/ucx_wrapper.cc:214]: minor: can you pls fix indentaiotn

File: test/apps/iodemo/ucx_wrapper.cc [hunk 3]
```diff
@@ -99,494 +102,613 @@ UcxContext::UcxContext(size_t iomsg_size, bool verbose) :
...
+    int port;
+
+    if (saddr->sa_family != AF_INET) {
+        return "<unknown address family>";
+    }
+
+    struct sockaddr_storage addr = {0};
+    memcpy(&addr, saddr, addrlen);
+    switch (addr.ss_family) {
+    case AF_INET:
```
Comment (yosefe) [inline test/apps/iodemo/ucx_wrapper.cc:230]: because of line 223, we can assume it's AF_INET

File: test/apps/iodemo/io_demo.cc [hunk 4]
```diff
@@ -83,50 +124,70 @@ class P2pDemoCommon : protected UcxContext {
...
+    inline void next_buffer() {
+        _cur_buffer_idx = (_cur_buffer_idx + 1) % _data_buffers.size();
+        assert(_cur_buffer_idx < opts().num_buffers);
+    }
+
+    inline size_t get_data_size() {
+        return opts().min_data_size +
+               (IoDemoRandom::rand() % static_cast<size_t>(
+                                            opts().max_data_size -
+                                            opts().min_data_size + 1));
```
Comment (yosefe) [inline test/apps/iodemo/io_demo.cc:156]: maybe do IoDemoRandom::rand(min, max) so IoDemoRandom::rand(opts().min_data_size , opts().max_data_size )?

**Pull request merged**